ecmascript/parser/src/parser/pat.rs

```diff
@@ -111,7 +111,7 @@ impl<'a, I: Tokens> Parser<'a, I> {
         }
 
         expect!(']');
-        let optional = self.input.syntax().dts() && eat!('?');
+        let optional = (self.input.syntax().dts() || self.ctx().in_declare) && eat!('?');
```

Comment:
Does around line 162 need to be updated to get rid of the syntax error? I think same in parser/expr.rs around 1230?

Reply:
Oh.. Thanks. I added bunch of error recovery logic, but I completely forgot about it.
(I'm too busy due to coronavirus...)